addOns/graphql/src/main/java/org/zaproxy/addon/graphql/GraphQlOptionsPanel.java

```diff
@@ -143,6 +104,94 @@ public void saveParam(Object obj) throws Exception {
         param.setRequestMethod((RequestMethodOption) getRequestMethodOptions().getSelectedItem());
     }
 
+    private JCheckBox getQueryGenEnabled() {
+        if (queryGenEnabled == null) {
+            queryGenEnabled =
+                    new JCheckBox(
+                            Constant.messages.getString("graphql.options.label.queryGenEnabled"),
+                            true);
+            queryGenEnabled.addItemListener(
+                    e -> {
+                        boolean selected = e.getStateChange() == ItemEvent.SELECTED;
+                        for (var c : getQueryGenConfigPanel().getComponents()) {
```

Comment:
This has the side effect of enabling `Additional Query Depth` even if `Lenient Maximum Query Depth` is not selected (worth noting that in the job dialogue they are not bound, guess this could be changed to be that way).